src/binding/transaction_log_file_posix.cpp

```diff
@@ -0,0 +1,97 @@
+#include "transaction_log_file.h"
+#include "macros.h"
+#include "util.h"
+
+#ifdef PLATFORM_POSIX
+
+namespace rocksdb_js {
+
+TransactionLogFile::TransactionLogFile(const std::filesystem::path& p, const uint32_t seq) :
+	path(p),
+	sequenceNumber(seq),
+	fd(-1)
+{}
+
+void TransactionLogFile::close() {
+	std::unique_lock<std::mutex> lock(this->fileMutex);
+
+	if (this->fd >= 0) {
+		DEBUG_LOG("%p TransactionLogFile::close Closing file: %s (fd=%d)\n",
+			this, this->path.string().c_str(), this->fd)
+		::close(this->fd);
+		this->fd = -1;
+	}
+}
+
+void TransactionLogFile::flushFile() {
+	if (this->fd < 0) {
+		return;
+	}
+
+	if (::fsync(this->fd) < 0) {
```

Comment:
I think we probably want to use `fdatasync`, a little faster and we don't need to update file metadata with every flush.

Reply:
Ah, yes `fdatasync()`. I'm fine with this, but we need to be comfortable with the tradeoffs. Since `fdatasync()` does not update the log file's mtime, the mtime will be stale until the kernel writeback thread runs and updates the file's mtime. Apparently this can take 5-30 seconds. Should the actualy OS panic or get powered off during this time, then we would have a stale mtime and the rocksdb-js code could potentially delete a transaction log outside the user-defined retention window. We default to 3 days, so odds of this being a problem are really slim. Just something to be aware of.

Reply:
And apparently macOS also doesn't support `fdatasync()`.